main.tf

```diff
@@ -39,10 +39,24 @@ resource "aws_route53_zone" "zone" {
 locals {
   records_expanded = [
     for record in var.records : merge({
-      record_id = join("-", compact([lower(record.type), lower(record.name)]))
-      ttl       = null
-      records   = null
-      alias     = null
+      # if no name is set for the record, we use the zones name
+      record_id = join("-", compact([lower(record.type), try(lower(record.name), ""),
```

Comment:
`""` will be removed by compact anyway... so not needed.

Reply:
my  bad... missed the try()

Reply:
true, resolved in 548c28a